### PR TITLE
added hasPrefix method and tests

### PR DESCRIPTION
**Pull request by rafapaezbas**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -418,6 +418,12 @@ class HyperBee {
     return iteratorToStream(new DiffIterator(new Batch(this, null, false, opts), new Batch(right, null, false, opts), opts), active)
   }
 
+  async hasPrefix (prefix) {
+    const ite = this.createRangeIterator({ gte: prefix })
+    await ite.open()
+    return ite.next()
```
Comment (mafintosh) [inline index.js:424]: capture the node here and check that the key actually starts with the prefix

Comment (mafintosh) [reply index.js:424]: ie, 'z' is gte 'a' but 'z' doesn't have 'a' as a prefix

Comment (rafapaezbas) [reply index.js]: corrected

File: index.js [hunk 2]
```diff
@@ -418,6 +418,13 @@ class HyperBee {
     return iteratorToStream(new DiffIterator(new Batch(this, null, false, opts), new Batch(right, null, false, opts), opts), active)
   }
 
+  async hasPrefix (prefix) {
```
Comment (mafintosh) [inline index.js:421]: @andrewosh do you have a better name for this?

File: index.js [hunk 3]
```diff
@@ -418,6 +418,13 @@ class HyperBee {
...
+    const ite = this.createRangeIterator({ gte: prefix })
```
Comment (mafintosh) [inline index.js:422]: @rafapaezbas add limit: 1 here also - that'll make it a bit faster

Comment (rafapaezbas) [reply index.js]: checkPrefix?

File: index.js (hunk 2 above)
Comment (rafapaezbas) [reply index.js:421]: checkPrefix ?

**Comment by mafintosh** [comment]
@rafapaezbas do we still need this?

**Comment by rafapaezbas** [comment]
Changes not needed anymore.

**Pull request closed without merging**